exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java

```diff
@@ -445,32 +446,47 @@ private static void writeScopeNameAndVersion(
   private static void writeAttributePairs(
       Writer writer, boolean initialComma, Attributes attributes) throws IOException {
     try {
-      attributes.forEach(
-          new BiConsumer<AttributeKey<?>, Object>() {
-            private boolean prefixWithComma = initialComma;
-
-            @Override
-            public void accept(AttributeKey<?> key, Object value) {
-              try {
-                if (prefixWithComma) {
-                  writer.write(',');
-                } else {
-                  prefixWithComma = true;
-                }
-                writer.write(NameSanitizer.INSTANCE.apply(key.getKey()));
-                writer.write("=\"");
-                writeEscapedLabelValue(writer, value.toString());
-                writer.write('"');
-              } catch (IOException e) {
-                throw new UncheckedIOException(e);
-              }
-            }
-          });
+      boolean prefixWithComma = initialComma;
+      for (Map.Entry<String, String> entry : sanitizeAttributePairs(attributes).entrySet()) {
+        try {
+          if (prefixWithComma) {
+            writer.write(',');
+          } else {
+            prefixWithComma = true;
+          }
+          writer.write(NameSanitizer.INSTANCE.apply(entry.getKey()));
+          writer.write("=\"");
+          writeEscapedLabelValue(writer, entry.getValue());
+          writer.write('"');
+        } catch (IOException e) {
+          throw new UncheckedIOException(e);
+        }
+      }
+      ;
     } catch (UncheckedIOException e) {
       throw e.getCause();
     }
   }
 
+  private static Map<String, String> sanitizeAttributePairs(Attributes attributes) {
+    Map<String, String> sanitizedAttributes = new HashMap<String, String>();
+    attributes.forEach(
+        new BiConsumer<AttributeKey<?>, Object>() {
+          @Override
+          public void accept(AttributeKey<?> key, Object value) {
```

Comment:
Nit: could be a lambda now that the `prefixWithComma` variable is a local, not a field

Reply:
done.